client/SDL/SDL3/sdl_clip.cpp

```diff
@@ -671,27 +703,51 @@ const void* sdlClip::ClipDataCb(void* userdata, const char* mime_type, size_t* s
 	}
 
 	{
-		HANDLE hdl[2] = { clip->_event, freerdp_abort_event(clip->_sdl->context()) };
-		auto status = WaitForMultipleObjects(ARRAYSIZE(hdl), hdl, FALSE, INFINITE);
-		if (status != WAIT_OBJECT_0)
-			return nullptr;
+		HANDLE hdl[2] = { freerdp_abort_event(clip->_sdl->context()), clip->_event };
+		DWORD status;
+		do
+		{
+			status = WaitForMultipleObjects(ARRAYSIZE(hdl), hdl, FALSE, 1 * 1000);
```

Comment:
please avoid polling loops

Reply:
Any reason for that ? We could get stuck forever there with `INFINITE` time, that also mean that we can handle a timeout for the SDL response as an improvement...

Reply:
@hardening yes, power consumption. makes a huge difference when I run `xfreerdp` without a lot of stuff going on. (well, not this one alone, but these polling loops keep accumulating)
if you get stuck fix the polling function ;)

Reply:
Hum, I'll change it, but AFAICT most polling loops in FreeRDP are on a model where they wait `INFINITE` on a real handle and a stop event.
If you have power consumption problems perhaps that's the sign that we have some polling loops that must have timeout (because of the real need of a timeout, or because we have an external event that can't be WaitForObjected) but the wait delay is not computed correctly. Could also be a missing `ResetEvent` somewhere..

Reply:
@hardening I don´t have problems (now), but if you use polling loops (as we did with `xfreerdp` previously) that increases `CPU` load and power consumption/wake ups
The loop here is also not exposing any of the arguments against `INFINITE`, it just waits for clipboard events or `rdpContext::abortEvent`